opacclient/libopac/src/main/java/de/geeksfactory/opacclient/apis/WebOpacAt.java

```diff
@@ -0,0 +1,290 @@
+            result.setCover(getCover(li));
+            final String statusImg = li.select(".status img").attr("src");
+            result.setStatus(statusImg.contains("-yes")
+                    ? SearchResult.Status.GREEN
+                    : statusImg.contains("-no")
+                    ? SearchResult.Status.RED
+                    : null);
+            results.add(result);
+        }
+        return new SearchRequestResult(results, totalResults, pageIndex);
+    }
+
+    static int parseTotalResults(final String navigation) {
+        final Matcher matcher = Pattern.compile("(von|of) (?<num1>\\d+)|(?<num2>\\d+) sonuçtan")
+                .matcher(navigation);
+        if (matcher.find()) {
+            final String num1 = matcher.group("num1");
+            return Integer.parseInt(num1 != null ? num1 : matcher.group("num2"));
+        } else {
+            return 0;
+        }
+    }
+
+    protected Map<String, String> buildSearchParams(List<SearchQuery> query)
+            throws IOException, JSONException {
+        final Map<String, String> params = new TreeMap<>();
+        if (query.size() == 1 && "q".equals(query.get(0).getSearchField().getId())) {
+            params.put("mode", "s");
+            params.put(query.get(0).getKey(), query.get(0).getValue());
+        } else {
+            int i = 0;
+            for (SearchQuery q : query) {
+                // crit_, value_, op_ are 0-indexed
+                params.put("crit_" + i, q.getKey());
+                params.put("value_" + i, q.getValue());
+                if (i > 0) {
+                    params.put("op_" + i, "AND");
+                }
+                i++;
+            }
+            params.put("mode", "a");
+            params.put("critCount", String.valueOf(params.size())); // 1-index
+        }
+        params.put("page", "1"); // 1-indexed
+        params.put("page_size", "30");
+        return params;
+    }
+
+    @Override
+    public DetailledItem getResultById(String id, String homebranch)
+            throws IOException, OpacErrorException {
+        final String html = httpGet(opac_url + "/search?view=detail&id=" + id, getDefaultEncoding());
+        final Document doc = Jsoup.parse(html);
+        final Element detailData = doc.select(".detailData").first();
+        final Element detailTable = detailData.select("table.titel").first();
+        final Element availabilityTable = doc.select(".bibliothek table").first();
+
+        final DetailledItem result = new DetailledItem();
+        final HashMap<String, String> copy = new HashMap<>();
+        result.addCopy(copy);
+        result.setId(id);
+        result.setCover(getCover(doc));
+        result.setTitle(detailData.select("h3").first().text());
+        result.setMediaType(MEDIA_TYPES.get(getCellContent(detailTable, "Medienart|Type of media")));
+        copy.put(DetailledItem.KEY_COPY_STATUS, getCellContent(availabilityTable, "Verfügbar|Available"));
```

Comment:
I agree that it's somewhat of a hack. ASAIK, only the languages `de`, `en`, `tr` are implemented at the moment. For "Verfügbar|Available", we could make use of the status image. But especially for "Medienart|Type of media" there are no indications. The only different approach would be via the search fields: "Medienart" is associated the key `ma`. This key could be used to determine the lookup term "Medienart" …Rather complex instead of specifying two/three translations here. :smile: